firewood/tests/db.rs

```diff
@@ -2,47 +2,28 @@
 // See the file LICENSE.md for licensing terms.
 
 use firewood::{
-    db::{BatchOp, Db as PersistedDb, DbConfig, DbError, WalConfig},
-    merkle::TrieHash,
+    db::{BatchOp, Db as PersistedDb, DbConfig, WalConfig},
+    v2::api::{self, Db, DbView, Proposal},
 };
 
-use std::{
-    collections::VecDeque,
-    fs::remove_dir_all,
-    ops::{Deref, DerefMut},
-    path::Path,
-    sync::Arc,
-};
+use std::{any::Any, collections::VecDeque, sync::Arc};
 
 // TODO: use a trait
 macro_rules! kv_dump {
     ($e: ident) => {{
         let mut s = Vec::new();
-        $e.kv_root_hash().unwrap();
-        $e.kv_dump(&mut s).unwrap();
+        let boxed_any: Box<&dyn Any> = Box::new(&$e);
+        boxed_any
+            .downcast_ref::<firewood::db::Db>()
+            .unwrap()
+            .kv_dump(&mut s)
+            .unwrap();
```

Comment:
This is gross... I see why you "have" to do it, but it doesn't make it any less gross... Something doesn't seem right here. We shouldn't have to hack together tests like this. `kv_dump` is already a hack and now we have a hack on a hack.